microbench/src/main/java/io/netty/microbench/handler/ssl/AbstractSslEngineBenchmark.java

```diff
@@ -0,0 +1,283 @@
+        };
+
+        abstract ByteBuffer newBuffer(int size);
+
+        void freeBuffer(ByteBuffer buffer) { }
+    }
+
+    @Param
+    public SslEngineProvider sslProvider;
+
+    @Param
+    public BufferType bufferType;
+
+    // Includes cipher required by HTTP/2
+    @Param({ "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256" })
+    public String cipher;
+
+    protected SSLEngine clientEngine;
+    protected SSLEngine serverEngine;
+
+    private ByteBuffer cTOs;
+    private ByteBuffer sTOc;
+    private ByteBuffer serverAppReadBuffer;
+    private ByteBuffer clientAppReadBuffer;
+    private ByteBuffer empty;
+
+    protected final void initEngines() {
+        clientEngine = newClientEngine();
+        serverEngine = newServerEngine();
+    }
+
+    protected final void destroyEngines() {
+        ReferenceCountUtil.release(clientEngine);
+        ReferenceCountUtil.release(serverEngine);
+    }
+
+    protected final void initHandshakeBuffers() {
+        cTOs = allocateBuffer(clientEngine.getSession().getPacketBufferSize());
+        sTOc = allocateBuffer(serverEngine.getSession().getPacketBufferSize());
+
+        serverAppReadBuffer = allocateBuffer(
+                serverEngine.getSession().getApplicationBufferSize());
+        clientAppReadBuffer = allocateBuffer(
+                clientEngine.getSession().getApplicationBufferSize());
+        empty = allocateBuffer(0);
+    }
+
+    protected final void destroyHandshakeBuffers() {
+        freeBuffer(cTOs);
+        freeBuffer(sTOc);
+        freeBuffer(serverAppReadBuffer);
+        freeBuffer(clientAppReadBuffer);
+        freeBuffer(empty);
+    }
+
+    protected final boolean doHandshake() throws SSLException {
+        clientEngine.beginHandshake();
+        serverEngine.beginHandshake();
+
+        SSLEngineResult clientResult = null;
+        SSLEngineResult serverResult = null;
+
+        boolean clientHandshakeFinished = false;
+        boolean serverHandshakeFinished = false;
+
+        do {
+            int cTOsPos = cTOs.position();
+            int sTOcPos = sTOc.position();
+
+            if (!clientHandshakeFinished) {
+                clientResult = clientEngine.wrap(empty, cTOs);
+                runDelegatedTasks(clientResult, clientEngine);
+                assert empty.remaining() == clientResult.bytesConsumed();
+                assert cTOs.position() - cTOsPos == clientResult.bytesProduced();
+
+                if (isHandshakeFinished(clientResult)) {
+                    clientHandshakeFinished = true;
+                }
+            }
+
+            if (!serverHandshakeFinished) {
+                serverResult = serverEngine.wrap(empty, sTOc);
+                runDelegatedTasks(serverResult, serverEngine);
+                assert empty.remaining() == serverResult.bytesConsumed();
+                assert sTOc.position() - sTOcPos == serverResult.bytesProduced();
+
+                if (isHandshakeFinished(serverResult)) {
```

Comment:
to avoid another conditional consider just assigning unconditionally

same comment as https://github.com/netty/netty/pull/6360/files#r102627897